pkg/account/deployer/client.go

```diff
@@ -0,0 +1,268 @@
+	if groupId != "" {
+		log.Debug("Trying to update policy with origin object ID (UUID) %q", groupId)
+		resp, err := d.client.GroupManagementAPI.EditGroup(ctx, d.accountInfo.AccountUUID, groupId).PutGroupDto(group).Execute()
+
+		if err = d.handleClientResponseError(resp, err, "unable to update policy with UUID: "+groupId); err != nil {
+			return "", err
+		}
+		return groupId, nil
+	}
+
+	result, resp, err := d.client.GroupManagementAPI.GetGroups(ctx, d.accountInfo.AccountUUID).Execute()
+	if err = d.handleClientResponseError(resp, err, "unable to get groups with name: "+group.Name); err != nil {
+		return "", err
+	}
+
+	// find groups with matching name
+	var existingGroups []accountmanagement.GetGroupDto
+	for _, g := range result.GetItems() {
+		if g.GetName() == group.Name {
+			existingGroups = append(existingGroups, g)
+		}
+	}
+
+	if len(existingGroups) == 0 {
+		var createdGroups []accountmanagement.GetGroupDto
+		createdGroups, resp, err = d.client.GroupManagementAPI.CreateGroups(ctx, d.accountInfo.AccountUUID).PutGroupDto([]accountmanagement.PutGroupDto{group}).Execute()
+		if err = d.handleClientResponseError(resp, err, "unable to create group with name: "+group.Name); err != nil {
+			return "", err
+		}
+		if len(createdGroups) < 1 {
+			return "", fmt.Errorf("unable to get UUID of created group with name: %s", group.Name)
+		}
+		return createdGroups[0].GetUuid(), nil
+	}
+
+	if len(existingGroups) > 1 {
+		log.Warn("Updating multiple policies with name %s", group.Name)
+	}
+
+	groupToUpdate := existingGroups[0]
+
+	resp, err = d.client.GroupManagementAPI.EditGroup(ctx, d.accountInfo.AccountUUID, groupToUpdate.GetUuid()).PutGroupDto(group).Execute()
+	if err = d.handleClientResponseError(resp, err, "unable to update group with name: "+group.Name); err != nil {
+		return "", err
+	}
+	return groupToUpdate.GetUuid(), nil
+}
+
+func (d *accountManagementClient) upsertUser(ctx context.Context, userId string) (remoteId, error) {
+	users, resp, err := d.client.UserManagementAPI.GetUsers(ctx, d.accountInfo.AccountUUID).Execute()
+	if err = d.handleClientResponseError(resp, err, "unable to get all users from account "+d.accountInfo.AccountUUID); err != nil {
+		return "", err
+	}
+	for _, u := range users.GetItems() {
+		if u.Email == userId {
+			return userId, nil
+		}
+	}
+
+	resp, err = d.client.UserManagementAPI.CreateUserForAccount(ctx, d.accountInfo.AccountUUID).UserEmailDto(accountmanagement.UserEmailDto{Email: userId}).Execute()
+	if err = d.handleClientResponseError(resp, err, "unable to create user with email: "+userId); err != nil {
+		return "", err
+	}
+
+	return userId, nil
+}
+
+func (d *accountManagementClient) updatePermissions(ctx context.Context, groupId string, permissions []accountmanagement.PermissionsDto) error {
+	if groupId == "" {
+		return fmt.Errorf("group id must not be empty")
+	}
+
+	if len(permissions) == 0 {
+		return nil
```

Comment:
Shouldn't we treat that as an error as well? When would this be correctly called with no permissions?

Reply:
don't have a strong opinion on that.. I just thought, if you give it no permission (which may happen or not) it does nothing. if you count that operation as an error or not is arguable. should i change it to err?

Reply:
agreed to keep it like it is